tests/integration-e2e/test_dry_run_workflow.py

```diff
@@ -131,3 +131,24 @@ def test_dry_run_workflow_multi():
         }
     }
     assert rank_allocation_standard == expected_items
+
+
+def test_dynamic_distribute():
+    """
+    Test that the dynamic distribute works
+    """
+
+    Path(("allocation_r1_c2.pkl.gz")).unlink(missing_ok=True)
+
+    from neurodamus import Neurodamus
+    config_file = str(SIM_DIR / "v5_sonata" / "simulation_config.json")
+    output_dir = str(SIM_DIR / "v5_sonata" / "output_coreneuron")
+    tmp_file = _create_tmpconfig_coreneuron(config_file)
+    GlobalConfig.verbosity = LogLevel.DEBUG
+
+    nd = Neurodamus(tmp_file.name,
+                    output_path=output_dir,
+                    num_target_ranks=1,
+                    modelbuilding_steps=2,
+                    lb_mode="Memory")
+    nd.run()
```

Comment:
Can we test for some result here (assert some distribution is happening) and maybe also add a better docstring?